### ISSUE TITLE
dataUrlToBlob ignores type parameters and assumes base64-encoding

### ISSUE THREAD

**Issue by Treora** [nolanlawson/blob-util#62]
As remarked [in #61](https://github.com/nolanlawson/blob-util/issues/61#issuecomment-672861241):

> a quick look at this library’s dataUrlToBlob() implementation reveals however that it throws away all of the media type parameters! And blindly assumes base64 encoding.

Current implementation: 

```
export function dataURLToBlob (dataURL: string): Blob {
  var type = dataURL.match(/data:([^;]+)/)[1]
  var base64 = dataURL.replace(/^[^,]+,/, '')

  var buff = binaryStringToArrayBuffer(atob(base64))
  return createBlob([buff], { type: type })
}
```

Issues:

- A data url without `;base64` appended to the media type should just percent-decode the data.

- The type should include its parameters; for example, in `text/plain;charset=iso-8859-5` (used with or without the base64-encoding), dropping the charset parameter might result in problems when decoding the blob’s data.

**Comment by nolanlawson** [comment]
Yep, this is true! I assumed base64 encoding for data URLs (for both input and output).
